### PR TITLE
improve: use bundle data client migrations

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR aims to utilize the migrated BundleDataClient and its supporting functions to remove code redundancy

### CODE REVIEW

File: src/clients/BundleDataClient.ts [hunk 1]
```diff
@@ -1,1285 +1,2 @@
...
-          assert(endTime >= startTime, "End time should be greater than start time.");
-          assert(startTime > 0, "Start time should be greater than 0.");
-          return [chainId, [startTime, endTime]];
-        })
-      ).filter(isDefined)
-    );
-  }
-}
+import { clients } from "@across-protocol/sdk";
+export class BundleDataClient extends clients.BundleDataClient.BundleDataClient {}
```
Comment (nicholaspai) [inline src/clients/BundleDataClient.ts:2]: Should we just export these out of src/clients/index?

File: src/dataworker/Dataworker.ts [hunk 2]
```diff
@@ -2280,16 +2278,18 @@ export class Dataworker {
...
-        latestMainnetBlock,
-        mainnetBundleEndBlock,
-        bundleV3Deposits,
-        bundleV3Fills,
-        bundleSlowFills,
-        unexecutableSlowFills,
-        expiredDepositsToRefundV3,
-        this.clients,
-        this.maxL1TokenCountOverride
+      this.rootCache[key] = Promise.resolve(
```
Comment (nicholaspai) [inline src/dataworker/Dataworker.ts:2281]: did you add Promise.resolve to make types match?

**Review by nicholaspai — COMMENTED** [review]
Mostly nitty comments except this question on the one larger change: https://github.com/across-protocol/relayer/pull/1778/files#r1735260728

Also, have you run integration tests?

File: src/dataworker/Dataworker.ts (hunk 2 above)
Comment (james-a-morris) [reply src/dataworker/Dataworker.ts:2281]: Yes - this is one of the few changes to logic that was needed to match types. `_buildPoolRebalanceRoot` is async in keyword only as there are no `awaits` or returns of promises. As a result - to keep the diff smaller I wrapped with a Promise.resolve( )

File: src/clients/BundleDataClient.ts (hunk 1 above)
Comment (james-a-morris) [reply src/clients/BundleDataClient.ts]: Done: https://github.com/across-protocol/relayer/pull/1778/commits/272b756f6f75cf6073dbfaecb3b6e1b39de230d5

File: src/dataworker/Dataworker.ts (hunk 2 above)
Comment (james-a-morris) [reply src/dataworker/Dataworker.ts]: A follow up: We don't need the Promise.resolve. https://github.com/across-protocol/relayer/pull/1778/commits/a7334a22cde693ff364fc537a22842e7b49cd7d0

**Comment by nicholaspai** [comment]
Can you integrate this SDK change as well: https://github.com/across-protocol/sdk/pull/715

**Comment by james-a-morris** [comment]
> Can you integrate this SDK change as well: [across-protocol/sdk#715](https://github.com/across-protocol/sdk/pull/715)

done

**Review by nicholaspai — APPROVED** [review]

**Pull request merged**